docs/intro-to-coalton.md

```diff
@@ -272,6 +272,41 @@ Type definitions introduce type constructors. For example, we may construct a so
 
 We'll see how to unpack these types using `match` later in this document.
 
+## Type Aliases
+
+Coalton allows the definition of parametric type aliases. Type aliases can be defined on primitive types and types created with `define-type` or `define-alias`.
+
+```lisp
+(coalton-toplevel
+  ;; New aliases are created with the DEFINE-ALIAS operator
+  (define-alias Coordinate Integer)
+  (define-alias (Pair :a) (Tuple :a :a))
+  (define-alias Translation (Pair Coordinate -> Pair Coordinate))
+  
+  (declare shift-right Translation)
+  (define (shift-right (Tuple x y))
+    (Tuple (1+ x) y))
+    
+  (define shifted-coordinate (shift-right (Tuple 0 0))))
+```
+
+Outside of a Coalton expression, `describe-type-of` displays the type of a symbol, including its aliases, and returns the type. `describe-alias` displays the alias along with its base type and returns the base type.
+
+```lisp
+COALTON-USER> shifted-coordinate
+#.(TUPLE 1 0)
+
+COALTON-USER> (type-of 'shifted-coordinate)
+(TUPLE INTEGER INTEGER)
+
+COALTON-USER> (describe-type-of 'shifted-coordinate)
```

Comment:
I think these functions and their output may need some workshopping.

Reply:
Do you have any feedback as far as what is good/bad about them?

Reply:
I already approved, but I'm not sure about the idea of showing the aliases in error messages anymore. It's maybe useful at first, but if you start stacking aliases it gets very unwieldy very fast:

```
    error: Type mismatch
      --> <macroexpansion>:4:2
        |
     4  |     (DEFINE (CREATE-ACCOUNT NAME INITIAL-BALANCE)
        |  ___^
     5  | |     (DO
     ...
     9  | |       (ACCOUNT-VALID? ACCOUNT)
     10 | |       (PURE ACCOUNT)))
        | |______________________^ Expected type '([ACCOUNTNAME := STRING] → [BALANCE := INTEGER] → ([BANKM := ((ENVT CONFIGURATION) [BANKSTATEM := (ST [BANKSTATE := (M:MAP [ACCOUNTNAME := STRING] ACCOUNT)])])] (RESULT STRING ACCOUNT)))' but got type '([ACCOUNTNAME := STRING] → [BALANCE := INTEGER] → ([BANKM := ((ENVT CONFIGURATION) [BANKSTATEM := (ST [BANKSTATE := (M:MAP [ACCOUNTNAME := STRING] ACCOUNT)])])] ACCOUNT))'
```